tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -420,6 +420,139 @@ def verify_failure_received():
         finally:
             lambda_client.delete_event_source_mapping(UUID=event_source_mapping_uuid)
 
+    @pytest.mark.parametrize(
+        "item_to_put1, item_to_put2, filter, calls",
+        [
+            # Test with no filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                None,
+                2,
+            ),
+            # Test with filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT"]},
+                1,
+            ),
+            # Test with OR filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT", "MODIFY"]},
+                2,
+            ),
+            # Test with 2 filters (AND), and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT"], "eventSource": ["aws:dynamodb"]},
+                1,
+            ),
+            # Test exists filter
+            (
+                {"id": {"S": "test123"}},
+                {"id": {"S": "test1234"}, "presentKey": {"S": "test123"}},
+                {"dynamodb": {"NewImage": {"presentKey": [{"exists": False}]}}},
+                1,
+            ),
+            # numeric filters
+            (
```

Comment:
the "numeric filters" and "prefix" did not trigger the lambda at all when running against AWS.

the json would look like this:
```
{
    "Records": [
        {
            "eventID": "f5d9470eb501e25133bb439ddbe7fec3",
            "eventName": "INSERT",
            "eventVersion": "1.1",
            "eventSource": "aws:dynamodb",
            "awsRegion": "us-east-1",
            "dynamodb": {
                "ApproximateCreationDateTime": 1660213902,
                "Keys": {
                    "id": {
                        "S": "test123"
                    }
                },
                "NewImage": {
                    "numericFilter": {
                        "N": "123"
                    },
                    "id": {
                        "S": "test123"
                    }
                },
                "SequenceNumber": "700000000008351815629",
                "SizeBytes": 34,
                "StreamViewType": "NEW_AND_OLD_IMAGES"
            },
            "eventSourceARN": "arn:aws:dynamodb:us-east-1:<accountid>:table/test-table-f7dc1c7b/stream/2022-08-11T10:31:38.042"
        }
    ]
}
```
So I wondered if the filter also needs the `"N"`. 
However, I found a [post on the AWS forum](https://repost.aws/questions/QUxxQDRk5mQ22jR4L3KsTkKQ/dynamo-db-streams-filter-with-nested-fields-not-working), where users where discussing the same issue - the filters were just not working as expected.

Maybe you could find out if this is indeed a problem on AWS side. It would be fine with me, if we leave the test and add the above link for documentation purpose as comment to the test cases.

Reply:
I don't have confirmation, but internally entries in DynamoDB are represented as strings (and no number). And tha'ts most likely the reason why numeric filters are not working. 
The result of this is that you can't filter with `numeric`, since there doesn't happen no conversion/cast from string to number.